### PR TITLE
Refactoring the support of the on-screen stars names

### LINKED ISSUE #1951: Invisibe labels of stars
Original question: https://answers.launchpad.net/stellarium/+question/698950

### Expected Behaviour
All labels of star are visible when "Labels and marks" in section Stars in far right position

### Actual Behaviour
Not all stars have visible labels on any zoom level when "Labels and marks" in section Stars in far right position

### PR DESCRIPTION
**Pull request by alex-w**
This pull request is introduced the additional config options to show extra designations as on-screen names of stars.

Fixes #1951 (issue)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gzotti — CHANGES_REQUESTED** [review]
In principle a nice idea. However, some things should be fixed before merge:

"old" double star designations: in contrast to "new" ones? Or in contrast to showing "no labels"?
It seems it currently is "old" or "none". If I check table C.9 in SUG C.4.2, I can see "old" (obsolete) ID and "new" (modern) ID listed, the difference is the ignorance of most modern scientists and/or computer/printing systems to use Greek (or even just lowercase!) letters. The numbers within the catalogs are still in wide use, just as the Greek letters for the catalogs have been used in typeset editions for many decades. These old catalog spellings are still widely known and may be easier for the human reader. So, I can imagine it is useful to show either "old" or "new" or "no" double star labels. In addition, table C.9 needs additions: OCC, BAS, KOH? (just look at TAU head) Others? 

Also, double star wins over var star label? Again, look at TAU head. SZ Tau is labelled when I select Var.stars, but only OCC926 when I select both Dbl. stars and Var.stars at the same time. And it has no label if "Designations" is disabled.  IMHO if both Dbl.&Var stars are selected, both labels should be shown, at danger of clutter. This would also hit the appearance of Bayer/Flamsted labels which may have to have double or var (or both) designations added. 

The next level would be a selection of which double star catalogs the user might be interested. But this may be another topic.

**Comment by alex-w** [comment]
> And it has no label if "Designations" is disabled. IMHO if both Dbl.&Var stars are selected, both labels should be shown, at danger of clutter.

Why? One star may have 3 designation from catalogs of doubles, one designation as variable star, plus Bayer and Flamesteed designation and one proper name, so, from your logic we should show all of them on the screen now?

**Comment by alex-w** [comment]
> "old" double star designations: in contrast to "new" ones? Or in contrast to showing "no labels"?

The modern designation of double stars start since WDS prefix :)

**Comment by gzotti** [comment]
> Why? One star may have 3 designation from catalogs of doubles, one designation as variable star, plus Bayer and Flamesteed designation and one proper name, so, from your logic we should show all of them on the screen now?

In fact, yes. If I check both double and var labels, and a star has both labels, still only one is shown. Why does the double star label "win" in this case? So, in this case we would need both labels. On Bayer/Flamsted I am less sure. Most (if not all) Bayer stars have Flamsted numbers, but tradition prefers Bayer over Flamsted where available. But if there are Bayer/Flamsted *and* the star has additional double or var star labels (or both? I don't have an example now), it can be reasoned to add those labels as well. 

The "use designations..." checkbox switches off the proper name as screen label, in favour of the 
( ((Bayer|Flamsted)[+Dbl][+Var]) | HIP). That's OK. 
 
> Or you want to see better description for checkboxes and in the SUG?

I don't understand the "old" part of the double star checkbox labels. These catalogs and names are still in use. The editors or typesetters of the latest edition of WDS seem to be unwilling or unable to print lowercase and Greek letters (?) so they declared the classical abbreviations obsolete, replacing them by 1-4 capital ASCII letters only. (OK, plus numbers and space for "H [1-6]") There is no informational difference between O&Sigma;22 and STT22, both can be identified from SUG Table C.9. So, being able to select between "old" or "new" WDS designations as labels for double stars would be nice. And we need additions in the SUG what  OCC, BAS, KOH (and maybe others) mean.

**Comment by gzotti** [comment]
WDS numbers may be more systematic, but are certainly less human readable. You could either say "old" (Struve etc) vs, "new" WD numbers (then we should label double stars with the WDS strings), or old and new catalog label spelling (as I understood previously). If you select "old" vs "WDS" labels, I'd wish to see the older spelling ("obsolete ID" in SUG Table C.9) for the "old" labels.

**Comment by alex-w** [comment]
> In fact, yes. 

No, no and non againg. Please do not mix on-screen labels and list of designation

> If I check both double and var labels, and a star has both labels, still only one is shown. 

Yes, this is correct behaviour. All objects in Stellarium has once label on the screen. From your logic you want to see on the screen something like: Altair (Atair) - α Aql - 53 Aql - GMC 5 - SMR 5 - DAL 27 - Σ II 10 - HIP 97649. This is terrible!

> Why does the double star label "win" in this case? 

By historical reasons - most double stars catalogs and their designations was created before catalog of variable stars was born.

> So, in this case we would need both labels. 

On the screen we should see only one label, which has high priority.

> On Bayer/Flamsted I am less sure. Most (if not all) Bayer stars have Flamsted numbers, but tradition prefers Bayer over Flamsted where available. But if there are Bayer/Flamsted and the star has additional double or var star labels (or both? I don't have an example now), it can be reasoned to add those labels as well.

See my example with Altair - you want to get an unreadable sky.

> The "use designations..." checkbox switches off the proper name as screen label, in favour of the
( ((Bayer|Flamsted)[+Dbl][+Var]) | HIP). That's OK.

That's not OK, because the full logic to show the designation is: Bayer|Flamsted|Dbl|Var|HIP

> I don't understand the "old" part of the double star checkbox labels. 

Just replace the "old" to "traditional" word

> These catalogs and names are still in use. The editors or typesetters of the latest edition of WDS seem to be unwilling or unable to print lowercase and Greek letters (?) so they declared the classical abbreviations obsolete, replacing them by 1-4 capital ASCII letters only. (OK, plus numbers and space for "H [1-6]") 

SIMBAD uses latinized designations for double stars also.

> There is no informational difference between OΣ22 and STT22, both can be identified from SUG Table C.9. So, being able to select between "old" or "new" WDS designations as labels for double stars would be nice. 

OΣ22 and STT22 is traditinal designations for double stars from Otto von Struve catalog. In the modern catalogs I saw both versions of designations. The really modern double stars has official WDS designation only.

> And we need additions in the SUG what OCC, BAS, KOH (and maybe others) mean.

OK

**Comment by gzotti** [comment]
> > If I check both double and var labels, and a star has both labels, still only one is shown.
> 
> Yes, this is correct behaviour. All objects in Stellarium has once label on the screen. From your logic you want to see on the screen something like: Altair (Atair) - α Aql - 53 Aql - GMC 5 - SMR 5 - DAL 27 - Σ II 10 - HIP 97649. This is terrible!

No, I never said this. The max would be "Bayer-Vxxx-STTyyy", which would likely still be shorter than a single WDS label. 
 
> > Why does the double star label "win" in this case?
> 
> By historical reasons - most double stars catalogs and their designations was created before catalog of variable stars was born.

This needs to be stated in the SUG then. 
 
> > So, in this case we would need both labels.
> 
> On the screen we should see only one label, which has high priority.

What if a user is more interested in var stars? 
 
> > On Bayer/Flamsted I am less sure. Most (if not all) Bayer stars have Flamsted numbers, but tradition prefers Bayer over Flamsted where available. But if there are Bayer/Flamsted and the star has additional double or var star labels (or both? I don't have an example now), it can be reasoned to add those labels as well.
> 
> See my example with Altair - you want to get an unreadable sky.

Definitely not. Just show what the user has asked to show with the checkboxes. 
 
> > The "use designations..." checkbox switches off the proper name as screen label, in favour of the
> > ( ((Bayer|Flamsted)[+Dbl][+Var]) | HIP). That's OK.
> 
> That's not OK, because the full logic to show the designation is: Bayer|Flamsted|Dbl|Var|HIP

OK. If you insist on just one designation, clearly say so in the SUG. Currently I can read a footnote saying "All of these items work independently from the parent option."

 
> > I don't understand the "old" part of the double star checkbox labels.
> 
> Just replace the "old" to "traditional" word

Yes. 

> OΣ22 and STT22 is traditinal designations for double stars from Otto von Struve catalog. In the modern catalogs I saw both versions of designations. The really modern double stars has official WDS designation only.

Currently I can select "old" (-->traditional") label or "no label". Would it be useful to show "WDS label" as alternative?

**Comment by alex-w** [comment]
> > Yes, this is correct behaviour. All objects in Stellarium has once label on the screen. From your logic you want to see on the screen something like: Altair (Atair) - α Aql - 53 Aql - GMC 5 - SMR 5 - DAL 27 - Σ II 10 - HIP 97649. This is terrible!
> 
> No, I never said this. The max would be "Bayer-Vxxx-STTyyy", which would likely still be shorter than a single WDS label.

Yes, you said this, because GMC 5 - SMR 5 - DAL 27 - Σ II 10 is traditional designations of double stars for Altair and from your logic and latest remark we should show α Aql - GMC 5 - SMR 5 - DAL 27 - Σ II 10. Do you see problem here? Stellarium always was show an one designation on the screen for any object - why you want to change this behaviour? How this change is related to the requirement of reduce labels on the screen?

> > By historical reasons - most double stars catalogs and their designations was created before catalog of variable stars was born.
> 
> This needs to be stated in the SUG then.

But this is obviously reason.


> > On the screen we should see only one label, which has high priority.
> 
> What if a user is more interested in var stars?

Who do not allow user to disable previous checkbox? What if a user is more interested in SAO designations?

> > See my example with Altair - you want to get an unreadable sky.
> 
> Definitely not. Just show what the user has asked to show with the checkboxes.

See changes in the code.

> > That's not OK, because the full logic to show the designation is: Bayer|Flamsted|Dbl|Var|HIP
> 
> OK. If you insist on just one designation, clearly say so in the SUG. 

This is standard behaviour in Stellarium (and probably in all planetariums). I understand your reasons but, in re-phrase, writing "open the box to eating pizza" on pizza box is a bit strange IMHO?

> Currently I can read a footnote saying "All of these items work independently from the parent option."

Yes, because enabling/disabling HIP designations is not touches visibility of common names or Bayer designations. Probably this is my bad English, but OK, I'll add an additional notes in the SUG.

> Currently I can select "old" (-->traditional") label or "no label". Would it be useful to show "WDS label" as alternative?

Why? Traditional labels was requested many times, but no WDS labels.

**Comment by gzotti** [comment]
> > > Yes, this is correct behaviour. All objects in Stellarium has once label on the screen. From your logic you want to see on the screen something like: Altair (Atair) - α Aql - 53 Aql - GMC 5 - SMR 5 - DAL 27 - Σ II 10 - HIP 97649. This is terrible!
> > 
> > 
> > No, I never said this. The max would be "Bayer-Vxxx-STTyyy", which would likely still be shorter than a single WDS label.
> 
> Yes, you said this, because GMC 5 - SMR 5 - DAL 27 - Σ II 10 is traditional designations of double stars for Altair and from your logic and latest remark we should show α Aql - GMC 5 - SMR 5 - DAL 27 - Σ II 10. Do you see problem here? Stellarium always was show an one designation on the screen for any object - why you want to change this behaviour? How this change is related to the requirement of reduce the number of labels on the screen?

There is certainly a common preference also in the zoo of popular double star catalogs. Just as Bayer wins over Flamsted, one double star catalog is more popular than the next, so there is no need to have GMC (what is this?), SMR (what is this?), DAL (what is this?) plus a Struve number as screen label.  AFAICS the three catalogs are not even given by SIMBAD and they are not listed in the SUG! (Please add these catalog names where they fit!)

We could say, no extra label if Bayer/Flamsted exists. But if an otherwise unlabelled star is both double and variable, and the user selected to show both, the label should reflect that.

> > > By historical reasons - most double stars catalogs and their designations was created before catalog of variable stars was born.
> > 
> > This needs to be stated in the SUG then.
> 
> But this is obviously reason.

Not really. 
 
> > > On the screen we should see only one label, which has high priority.
> > 
> > What if a user is more interested in var stars?
> 
> Who do not allow user to disable previous checkbox? What if a user is more interested in SAO designations?

SAO (like HR, HD, GSC, FK5, ...) is not a current option. 

> > OK. If you insist on just one designation, clearly say so in the SUG.
> 
> This is standard behaviour in Stellarium (and probably in all planetariums). I understand your reasons but, in re-phrase, writing "open the box to eating pizza" on pizza box is a bit strange IMHO?
> 
> > Currently I can read a footnote saying "All of these items work independently from the parent option."
> 
> Yes, because enabling/disabling HIP designations is not touches visibility of common names or Bayer designations. Probably this is my bad English, but OK, I'll add an additional notes in the SUG.

Then the extra options *depend* on the parent! And var *depends* on being non-double, etc. The dependence rules need to be given. 

> > Currently I can select "old" (-->traditional") label or "no label". Would it be useful to show "WDS label" as alternative?
> 
> Why? Traditional labels was requested many times, but no WDS labels.

Then simply call the flag "double star labels" not "old double star labels". Just that "old" implies a "new" alternative. Maybe add "traditional" in the tooltip.

**Comment by alex-w** [comment]
> We could say, no extra label if Bayer/Flamsted exists. But if an otherwise unlabelled star is both double and variable, and the user selected to show both, the label should reflect that.

I disagree, because we should always show both Bayer and Flamesteed designations if their existing to complete follow your proposal.

**Comment by alex-w** [comment]
> Then the extra options depend on the parent! And var depends on being non-double, etc. The dependence rules need to be given.

How it possible? The parent option enabled/disabled show common names for stars. Where you see the dependence for double star designation or HIP number?

**Comment by gzotti** [comment]
Uh. I just was about to commit the updated documentation and fixed tooltip strings... If you want your way, you can revert your last commit. Else I will have to cleanup a conflict. The major point of misunderstanding was that your implementation (sequence with preference to double stars where available) did not fit your documentation ("append").

**Comment by alex-w** [comment]
> There is certainly a common preference also in the zoo of popular double star catalogs. Just as Bayer wins over Flamsted, one double star catalog is more popular than the next, so there is no need to have GMC (what is this?), SMR (what is this?), DAL (what is this?) plus a Struve number as screen label. AFAICS the three catalogs are not even given by SIMBAD and they are not listed in the SUG! (Please add these catalog names where they fit!)

GMC = G. Micello; SMR = J. Schlimmer; DAL = J.A. Daley - this is designations of observers. Struve compounded their observations as catalog. The observers from your list is not created a catalogs, but their observations was published as lists in various journals, like Nature or Pub. Ludwig Schupmann Obs. or throug private communication with creators of WDS catalog. For example you ask what meaning OCC designation, so, let's see references for OCC designation in WDS:
OCC  OCC1971   O   j  Taylor, G.E., O'Leary, B., van Fladern, T.C. Bartholdi, P., Owen, F.,               1971Natur.234..405T
OCC  OCC1971          Hubbard, W.B., Smith, B.A., Smith, S.A., Fallon, F.W. Devinney, E.J. & Oliver, J
OCC  OCC1971          Nature 234, 405, 1971             (also summarized in OCC2010)
OCC  OCC1972   O   j  Hubbard, W.B. & van Flandern, T.C.                                                  1972AJ.....77...65H
OCC  OCC1972          AJ 77, 65, 1972
OCC  OCC1976   O   j  Taylor, G.E., Walker, E.N., Quintanilla, A.R. & Lopez de Coca, M.P.                 1976A&A....50..121T
OCC  OCC1976          A&A 50, 121, 1976                 (also summarized in OCC2010)
OCC  OCC1990   O   j  Dunham, D.W., Dunham, J.B., Binzel, R.P., Evans, D.S., Freuh, M., Henry, G.W.,      1990AJ.....99.1636D
OCC  OCC1990          A'Hearn, M.A., Schnurr, R., Betts, R., Haynes, H., Orcutt, R., Bowell, E., et al
OCC  OCC1990          AJ 99, 1636, 1990                 (also summarized in OCC2010)
OCC  OCC1993   O   j  Sato, I., Soma, M. & Hirose, T.                                                     1993AJ....105.1553S
OCC  OCC1993          AJ 105, 1553, 1993 (also summarized in OCC2010)
OCC  OCC2001   O   -  Vasundhara, R., Jayakumar, K., Velu, C., Appakutty, M., Paranjpye, A.,              2001BASI...29..577V
OCC  OCC2001          Kulkarni, H., Nitsure, P., Prabhunne, M. & Shah, K
OCC  OCC2001          Bull. Astr. Soc. India 29, 577, 2001  (also summarized in OCC2010)
OCC  OCC2002   O   -  Occult. Doubles Cat., D. Dunham, ed
OCC  OCC2002          http://www.lunar-occultations.com/iota/iotandx.htm
OCC  OCC2008   O   j  Ling, A.                                                                            2008JRASC.102...15L
OCC  OCC2008          JRASC 102, 15, 2008
OCC  OCC2009   O   u  Herald, D. and Sandy, R.                                                            2009JDSO....5..208H
OCC  OCC2009          JDSO 5, 208, 2009
OCC  OCC2009b  O   j  Lin, C.-L., Zhang, Z.-W., Chen, W.P., King, S.-K., Lin, H.-C., Bianco, F.B.,        2009PASP..121..359L
OCC  OCC2009b         Lehner, M.J., Coehlo, N., Wang, J.-H., Mondal, S., Alcock, C., Axelrod, T., et al
OCC  OCC2009b         PASP 121, 359, 2009               (also summarized in OCC2010)
OCC  OCC2010   O   u  Herald, D., Boyle, R., Dunham, D., Hirose, T., Maley, P., Timerson, B., Farris,     2010JDSO....6...88H
OCC  OCC2010          T., Frappa, E., Lecacheux, J., Hayamizu, T., Kozubal, M., Nolthenius, R., et al
OCC  OCC2010          JDSO 6, 88, 2010
OCC  OCC2010b  O   u  Herald, D., Loader, B. & Parker, S.                                                 2010JDSO....6..150H
OCC  OCC2010b         JDSO 6, 150, 2010
OCC  OCC2010c  O   -  Herald, D., Kashiwagura, M., Loader, B., Messner, S., Sandy, R. & Togashi, H.       2010JDSO....6..152H
OCC  OCC2010c         JDSO 6, 152, 2010
OCC  OCC2010d  O   u  Loader, B., Asada, Y., Breit, D., Bradshaw, J., Gault, D., Herald, D., Iverson,     2010JDSO....6..176L
OCC  OCC2010d         E., Ishida, M., Karasaki, H., Kashiwagura, M., Miyashita, K.,  Messner, S., et al
OCC  OCC2010d         JDSO 6, 176, 2010
OCC  OCC2011   O   u  George, T., Timerson, B., Coughlin, K. & Fleishman, R.                              2011JDSO....7..129G
OCC  OCC2011          JDSO 7, 129, 2011
OCC  OCC2011b  O   u  Loader, B. Bradshaw, J., Breit, D., Gault, D., George, T., Herald, D., Iverson,     2011JDSO....7..159L
OCC  OCC2011b         E., Ishida, M., Karasaki, H., Kashiwagura, M., Kenmotsu, K., Lowe, D., et al
OCC  OCC2011b         JDSO 7, 159, 2011
OCC  OCC2011c  O   u  George, T., Timerson, B., Cooke, B., Degenhardt, S., Dunham, D.W., Messner, S.,     2011JDSO....7..169G
OCC  OCC2011c         Suggs, R., Venable, R. & Warren, W.H
OCC  OCC2011c         JDSO 7, 169, 2011
OCC  OCC2011d  O   -  George, T., Timerson, B., Beard, T., Blank, T., Dantowitz, R., Davis, J.,           2011JDSO....7..175G
OCC  OCC2011d         di Cicco, D., Dunham, D.W., Hill, M., Sliski, A. & Sumner, R
OCC  OCC2011d         JDSO 7, 175, 2011
OCC  OCC2011e  O   j  Bulder, H., Loader, B., Herald, D., Messner, S., Ling, A., Raab, H.,
OCC  OCC2011e         Culbertson, B., Black, S., Ishida, M., Ohno, T., & Miyashita, K
OCC  OCC2011e         J. for Occultation Astron., #3, Jul-Sep 2011     (some data also in OCC2008)
OCC  OCC2012a  O   j  George, T., Timerson, B., Degenhardt, S., Bracken, C., Cadmus, R., Carcich, A.,     2012JDSO....8..306G
OCC  OCC2012a         Centala, J., Modic, R., & Slauson, D
OCC  OCC2012a         JDSO 8, 306, 2012
OCC  OCC2012b  O   u  Loader, B., Asada, Y., Bradshaw, J., Breit, D., Fuertes, J., Gault, D., George,T.,  2012JDSO....8..313L
OCC  OCC2012b         Graham, P., Herald, D., Holenstein, B., Iverson, E., Ishida, M., et al
OCC  OCC2012b         JDSO 8, 313, 2012
OCC  OCC2013a  O   -  George, T., Timerson, B., Brooks, J., Conrad, S., Dunham, J.B., Dunham, D.W.,       2013JDSO....9...88G
OCC  OCC2013a         Jones,R., Lipka, T.R., Thomas, W., Warren, W.J., Jr., Wasson, R. & Wisniewski, J
OCC  OCC2013a         JDSO 9, 88, 2013
OCC  OCC2013b  O   j  Giacchini, B.L., Timerson, B., George, T., Degenhardt, S. & Herald, D.              2013JDSO....9..130G
OCC  OCC2013b         JDSO 9, 130, 2013
OCC  OCC2013c  O   j  Casas, R., Juan, J., Naves, R., Perello, C., Rovira, J., Selva, A. & Schnabel, C.   2013JDSO....9..183C
OCC  OCC2013c         JDSO 9, 183, 2013
OCC  OCC2013d  O   j  Herald, D., Talbot, J., Bradshaw, J., Broughton, J., Hanna, W. & Watson, D.         2013JDSO....9..226H
OCC  OCC2013d         JDSO 9, 226, 2013
OCC  OCC2013e  O   -  Herald, D.                                                                          2013JDSO....9..274H
OCC  OCC2013e         JDSO 9, 274, 2013
OCC  OCC2014a  O   -  Loader, B.                                                                          2014JDSO...10....2L
OCC  OCC2014a         JDSO 10, 2, 2014
OCC  OCC2014b  O   -  Gault, D.                                                                           2014JDSO...10....4G
OCC  OCC2014b         JDSO 10, 4, 2014
OCC  OCC2014c  O   -  Herald, D., Talbot, J., Chad, C., Broughton, J. & Anderson, P.                      2014JDSO...10...29H
OCC  OCC2014c         JDSO 10, 29, 2014
OCC  OCC2014d  O   j  Herald, D., Talbot, J. & Kerr, S.                                                   2014JDSO...10..107H
OCC  OCC2014d         JDSO 10, 107, 2014
OCC  OCC2014e  O   u  Loader, B., Bradshaw, J., Breit, D., Edens, E., Forbes, M., Gault, D., George, T.,  2014JDSO...10..150L
OCC  OCC2014e         Haymes, T., Herald, D., Holenstein, B., Ito, T., Iverson, E., et al
OCC  OCC2014e         JDSO 10, 150, 2014
OCC  OCC2014f  O   -  Soma, M., Hayamizu, T., Ishida, M., Owada, M., Ida, M., Aikawa, R., Hashimoto, A.,  2014JDSO...10..240S
OCC  OCC2014f         Horaguchi, T., Kitazaki, K., Uchiyama, S., Uehara, S., Yaeza, A., et al
OCC  OCC2014f         JDSO 10, 240, 2014
OCC  OCC2016a  O   j  Loader, B., Asai, A., Bardecker, J., Bradshaw, J., Breit, D., Buettner, D.,         2016JDSO...12..143L
OCC  OCC2016a         Chad, C., Denyer, P., Edens, E., Forbes, M., Gault, D., George, T., et al
OCC  OCC2016a         JDSO 12, 143, 2016
OCC  OCC2016b  O   -  Gault, D. & Herald, D.                                                              2016JDSO...12..556G
OCC  OCC2016b         JDSO 12, 556, 2016
OCC  OCC2017a  O   j  Loader, B., Anderson, P., Asai, A., Bradshaw, J., Breit, D., Forbes, M.,            2017JDSO...13..585L
OCC  OCC2017a         Gault, D., George, T., Haymes, T., Herald, D., Ishida, M., Kerr, S., et al
OCC  OCC2017a         JDSO 13, 585, 2016
OCC  OCC2018a  O   j  Timerson, B., George, T., Blank, T., Maley, P., Messner, S. & Moore, J.             2018JDSO...14..395T
OCC  OCC2018a         JDSO 14, 395, 2018
OCC  OCC2019a  O   j  Timerson, B., George, T., Bean, J., Blank, T., Bardecker, J., Dunham, D., Dunham,   2019JDSO...15..162T
OCC  OCC2019a         J., Gimple, B., Maley, P., Anthony, S., Wiesenborn, C., Hunt, T., Sumner, R. &
OCC  OCC2019a         Thomas, W
OCC  OCC2019a         JDSO 15, 162, 2019
OCC  OCC2019b  O   j  Hirose, T., Ida, M., Yamamura, H., Uchiyama, S., Hashimoto, A., Aikawa, R.,         2019JDSO...15..268H
OCC  OCC2019b         Terakubo, K., Sato, M. & Shima, K
OCC  OCC2019b         JDSO 15, 268, 2019
OCC  OCC2019c  O   j  Loader, B., Asada, Y., Asai, A., Bardecker, J., Bourgeois, J., Bradshaw, J., Chad,  2019JDSO...15..503L
OCC  OCC2019c         C., Ellington, C., Gault, D., Giacchini, B., Haymes, T., Herald, D., Ishida, M.,
OCC  OCC2019c         Iverson, E., Kerr, S., Manek, J., McKay, G., Messner, S., Pratt, A., Sandy, R.,
OCC  OCC2019c         Watanabe, H., Yamamura, H. & Yoshida, H
OCC  OCC2019c         JDSO 15, 503, 2019
OCC  OCC2019d  O   j  Giacchini, B.L., George, T., Spagnotto, J. & Wilberger, A.J.                        2019JDSO...15..554G
OCC  OCC2019d         JDSO 15, 554, 2019
OCC  OCC2020a  O   -  Perello, C., Frappa, E., Janik, T., Kattentidt, B., Polak, J., Rottenborn, M. &     2020JDSO...16...87P
OCC  OCC2020a         Selva, A
OCC  OCC2020a         JDSO 16, 87, 2020
OCC  OCC2020b  O      Herald, D.                                                                          2020JDSO...16..113H
OCC  OCC2020b         JDSO 16, 105, 2020
OCC  OCC2020c  O      Bardecker, J. & Messner, S.                                                         2020JDSO...16..405B
OCC  OCC2020c         JDSO 16, 405, 2020
OCC  OCC2020d  O      Gault, D., Barry, T., Nosworthy, P., Hanna, W., Messner, S., Oesper, D., Dunford,   2020JDSO...16..440G
OCC  OCC2020d         R., Trank, R., Preston, S. & Herald, D
OCC  OCC2020d         JDSO 16, 440, 2020
OCC  OCC2020e  O      Boutet, M. & Sanchez, J.                                                            2020JDSO...16..483B
OCC  OCC2020e         JDSO 16, 483, 2020
OCC  OCC2021a  O      Hayamizu, T., Soma, M., Watanabe, Ha., Asai, A., Watanable, Hi. & Nohara, H.        2021JDSO...17..131H
OCC  OCC2021a         JDSO 17, 131, 2021
OCC  OCC2021b  O      Frappa, E., Zeleny, P., Baruffetti, P., Bendjeddou, A., Bigi, M., Bouazara, O.,     2021JDSO...17..230F
OCC  OCC2021b         Gacem, A., Khenifer, H.M., Mokhtari, L., Nosal, P., Rayane, H., Selimi, T.E.M.,
OCC  OCC2021b         Sposetti, S. & Aissa, D.B
OCC  OCC2021b         JDSO 17, 230, 2021

What I should write in the SUG for this designation?

**Comment by alex-w** [comment]
> Uh. I just was about to commit the updated documentation and fixed tooltip strings... If you want your way, you can revert your last commit. Else I will have to cleanup a conflict. The major point of misunderstanding was that your implementation (sequence with preference to double stars where available) did not fit your documentation ("append").

The last commit is removed

**Comment by gzotti** [comment]
OCC: A difficult case then. What does the name stand for? Discovery by Occultations? I have really no idea!

If OCC is used in the WDS catalog, the entry may be:

OCC ... various sources collected as OCC in the WDS.

**Comment by alex-w** [comment]
@gzotti please check tooltips

**Comment by alex-w** [comment]
I think this is ready to merge now...

**Comment by alex-w** [comment]
@gzotti should we sort the list of designations in the SUG (C.6) by asc. in second column?

**Comment by gzotti** [comment]
Yes, alphabetic sorting is a good idea. Maybe even make a horizontal line between classic catalogs and your new WDS observer tags. I have no knowledge which are the "catalogs" (single publications) or "observer entries in WDS lists" (like those tagged OCC).

**Comment by alex-w** [comment]
> Yes, alphabetic sorting is a good idea. 

OK

> Maybe even make a horizontal line between classic catalogs and your new WDS observer tags. I have no knowledge which are the "catalogs" (single publications) or "observer entries in WDS lists" (like those tagged OCC).

This is really problematic, because collection of few lists is have no difference with catalog (one big list).

File: guide/app_star_catalogue.tex [hunk 1]
```diff
@@ -1218,73 +1218,157 @@ \subsection{Designations}
...
+---		        	& HDS  	& ** & \emph{Hipparcos Catalog} \\
+AC		        	& AC  	& ** & A. Clark \\
+AG		        	& AG  	& ** & \emph{Astronomische Gesellschaft Katalog} \\
+AGC		        	& AGC 	& ** & A.G. Clark \\
+Arg		        	& ARG 	& ** & F.W.A. Argelander \\
+All		        	& ALL 	& ** & R.M. Aller \\
+Dju  	        	& DJU 	& ** & P. Djurkovic \\
+Don  	        	& DON 	& ** & H.F. Donner \\
+Doo  	        	& DOO 	& ** & E. Doolittle \\
+Erk  	   			& FRK 	& ** & W.S. Franks \\
```
Comment (gzotti) [inline guide/app_star_catalogue.tex:1272]: Typo? **E**rk vs. **F**RK?

Comment (alex-w) [reply guide/app_star_catalogue.tex:1272]: Yes, typo

File: guide/app_star_catalogue.tex [hunk 2]
```diff
@@ -1218,73 +1218,157 @@ \subsection{Designations}
...
+Se  	 	        & SE  	& ** & A. Secchi \\
+Sei   		        & SEI 	& ** & J. Scheiner \\
+Smyth  	 	        & SMY 	& ** & W.H. Smyth \\
+Vou  	        	& VOU 	& ** & J.G.E.G. Voute \\
+Webb  		        & WEB 	& ** & T.W. Webb \\
+Wnc  	        	& WNC 	& ** & F.A. Winnecke \\
+---		        	& UC  	& ** & \emph{USNO CCD Astrographic Catalog } \\
+Skf		        	& SKF  	& ** & B.A. Skiff \\
+---		        	& TDS, TDT 	& ** & \emph{Tycho Double Star} \\
+Tod		        	& TOI  	& ** & Stassun et al. 		\\
```
Comment (gzotti) [inline guide/app_star_catalogue.tex:1289]: Typo or real? To**d** vs. TO**I**? And how does Stassun fit to to[di]? (It may be OK, but please double-check!)

Comment (alex-w) [reply guide/app_star_catalogue.tex:1289]: Real

File: guide/app_star_catalogue.tex [hunk 3]
```diff
@@ -1218,73 +1218,157 @@ \subsection{Designations}
...
+Soz		        	& SOZ 	& ** & R.-D. Scholz \\
+Bko		        	& BKO 	& ** & E. Berko \\
+---		        	& BPM 	& ** & \emph{Bordeaux PM2000 catalogue} \\
+Mil		        	& MIL 	& ** & J.A. Miller \\
+Tob		        	& TOB 	& ** & T. Tobal  \\
+Scj		        	& SCJ 	& ** & H. Schjellerup \\
+Dea		        	& DEA 	& ** & N.R. Deacon \\
+Bvd		        	& BVD 	& ** & R. Benavides \\
+---		        	& RAO 	& ** & \emph{Robo AO project} \\
+Lv		        	& LV 	& ** & F.P. Leavenworth \\
```
Comment (gzotti) [inline guide/app_star_catalogue.tex:1345]: tabulation?

**Comment by gzotti** [comment]
In the older list you had approx. year ranges for what I think are the classical "catalogs". Probably you can add these years back, where available. There will be no new Struwe numbers, but OCC may still be added.

File: src/core/modules/StarMgr.cpp [hunk 4]
```diff
@@ -433,6 +446,9 @@ void StarMgr::init()
 	setFlagLabels(conf->value("astro/flag_star_name",true).toBool());
 	setFlagAdditionalNames(conf->value("astro/flag_star_additional_names",true).toBool());
 	setDesignationUsage(conf->value("astro/flag_star_designation_usage", false).toBool());
+	setFlagDblStarsDesignation(conf->value("astro/flag_star_designation_dbl_stars", false).toBool());
```
Comment (gzotti) [inline src/core/modules/StarMgr.cpp:449]: I think we can shorten the names to flag_star_designation_dbl

Comment (alex-w) [reply src/core/modules/StarMgr.cpp]: Good idea, thanks!

**Comment by alex-w** [comment]
Comment to the SUG: the section about designation of double stars maybe improve, but only after reading the special literature.

**Comment by gzotti** [comment]
Great list! Testing a bit...

**Comment by gzotti** [comment]
Fine for me!

**Pull request merged**